storm-core/src/jvm/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.scheduler.blacklist;
+
+import org.apache.storm.Config;
+import org.apache.storm.metric.StormMetricsRegistry;
+import org.apache.storm.scheduler.Cluster;
+import org.apache.storm.scheduler.IScheduler;
+import org.apache.storm.scheduler.SupervisorDetails;
+import org.apache.storm.scheduler.Topologies;
+import org.apache.storm.scheduler.WorkerSlot;
+import org.apache.storm.scheduler.blacklist.reporters.IReporter;
+import org.apache.storm.scheduler.blacklist.strategies.IBlacklistStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+public class BlacklistScheduler implements IScheduler {
+    private static final Logger LOG = LoggerFactory.getLogger(BlacklistScheduler.class);
+    IScheduler underlyingScheduler;
+    @SuppressWarnings("rawtypes")
+    private Map _conf;
+
+    protected int toleranceTime;
+    protected int toleranceCount;
+    protected int resumeTime;
+    protected IReporter reporter;
+    protected IBlacklistStrategy blacklistStrategy;
+
+    protected int nimbusMonitorFreqSecs;
+
+    protected Map<String, Set<Integer>> cachedSupervisors;
+
+    //key is supervisor key ,value is supervisor ports
+    protected CircularBuffer<HashMap<String, Set<Integer>>> badSupervisorsTolerance;
+    protected Set<String> blacklistHost;
+
+    public BlacklistScheduler(IScheduler underlyingScheduler) {
+        this.underlyingScheduler = underlyingScheduler;
+    }
+
+    @Override
+    public void prepare(Map conf) {
+        LOG.info("prepare black list scheduler");
+        LOG.info(conf.toString());
+        underlyingScheduler.prepare(conf);
+        _conf = conf;
+        if (_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME)) {
+            toleranceTime = (Integer) _conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME);
```

Comment:
Utils.getInt is done.
but I don't think it's a good idea that I set nimbusMonitorFreqSecs to 10 by default here I think change a bit in nimbus_test.clj is better.